### ISSUE TITLE
Inclusion of video lessons and/or asciinema-casts

### ISSUE THREAD

**Issue by ybur-yug** [doomspork/elixir-school#551]
## The Idea
[Asciinema](http://asciinema.org) is a wonderful tool for recording terminals.
I have worked with it in the past and find it quite a pleasure to use and it has a ton of neat features. You can even copy/paste the output its giving as the 'video' plays.

I'd love to have a screencast-of-sorts for each lesson.

### Straight Asciinema Cast
The first/most obvious is just doing asciinema casts and then writing comments as you go through the code you are doing that will give insights that go along with the lesson. You can see a sample [here](https://asciinema.org/a/69i2tymxacga62pizlghfooqs) to get exactly what I mean. The files that make it up are all JSON so it could also just have the comments/content edited to give translated examples. The hosting is also trivially simple and free.

### Asciinema Cast With Audio
This is what I find to be the most appealing. If we look at the filesize of the asciinema cast shared prior it was around 200kb. This with an audio file would probably pop in at under 1mb, and makes it nice and heavily available and easy to get. 

However, doing audio, and losing the commenting that explains things, makes translation require an entire new track (and thats if and only if the code is reasonable for another language to explain while being typed out in english, which can be hard)

### Full-On Screencast
This is the nuclear option and requires time money and extra editing effort. I don't see the point myself with them all being freely available because it would take a profit motive for me to give this style the proper attention. Maybe someone else is more altruistic, though.

I'm going to record some audio for another screencast today and get a sample for that up and will add it. Does anyone have thoughts?

**Comment by doomspork** [comment]
@ybur-yug / @brain-geek / @pragmaticivan / @setzer777 / @pjhampton  — How do you all feel about this?

**Comment by nscyclone** [comment]
The idea is great and I thought about this earlier. I suppose that this is really the most modern, convenient and fastest form of learning something at the moment. But as @ybur-yug already mentioned it is also a quite expensive and hard work. Probably this means that some part of the content should become non-free and go beyond the open-source format.
We could also start another Twitter-poll once we think up the concept of this. This may seem a good idea for us but no one can guarantee that we think alike with the majority of community. 
Also there is already an [existing course](https://www.pluralsight.com/courses/meet-elixir) made by José Valim.

I'd like to help with translations, voice-overs and anything else if needed.

**Comment by brain-geek** [comment]
We can try some on either basic or new lessons with Asciinema 2 gif - this shouldn't be very hard, but will help validating the idea.

**Comment by ybur-yug** [comment]
We wouldnt even need to convert it to GIF I dont believe @brain-geek.
Since the casts are just JSON streamed into a window, in the worst case we could make a private site to display them and iframe it in. Gif adds a TON of filesize overhead we dont need/want. Raw Asciinema files clock in at around 100kb/minute so the bandwidth cost would be super low to get everything out. This also would remove the ability to copy/paste from the 'video' itself, which I think is a huge benefit and a lot of screencasts have the 'ascii' written up version, but IMHO being able to copy the code in the video directly is a huge win.

**Comment by brain-geek** [comment]
@ybur-yug But overall - what do you think about adding few short examples to an existing lesson as an example?

For example, [iex interaction start](https://elixirschool.com/lessons/basics/basics/#trying-interactive-mode) could use it.

**Comment by ybur-yug** [comment]
@brain-geek I totally think that is an awesome start. And its far too huge an undertaking to consider this in one sweep.

Maybe this would be a sane plan to begin:

1. Make an issue with a list of lessons
2. Have a sublist for each one of potential quick examples
3. Handle the PR's adding them like we do translations right now
4. Start getting feedback and see what the users think

**Comment by yakschuss** [comment]
@ybur-yug hey Bobby, I'm fairly new to Elixir but I'd love to pair on something like this if you're interested.

**Comment by ybur-yug** [comment]
@yakschuss I had forgotten about this one being open. I'm not quite sure pairing would work well on asciicasts just because the nature of it is being a very tight, perfected bit that is done with relative quickness. However, I'm sure we could think of something that could be good to hack on together in the near future. DM me on slack or email me and we could brainstorm a bit. I should be doing some work on the Phoenix Guides here soon, too, I think. Waiting on the transition to go from being separate to them being merged into the Phoenix repo itself.
